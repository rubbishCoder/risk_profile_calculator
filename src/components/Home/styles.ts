import { StyleSheet } from 'react-native';
import Colors from '../../common/Colors';
import RV from '../../common/utils/RV';

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: Colors.white,
        justifyContent: 'center',
        alignItems: 'center'
    },
    textContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: Colors.blue2,
        width: '90%',
        height: RV(300),
        borderRadius: RV(50)
    },
    text: {
        color: Colors.white,
        fontSize: RV(24),
        textAlign: 'center',
        padding: RV(16),
        fontWeight: 'bold'
    },
    buttonContainer: {
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: Colors.white,
        paddingVertical: RV(16),
        paddingHorizontal: RV(20),
        borderRadius: RV(12)
    },
    buttonText: {
        fontSize: RV(24),
        textAlign: 'center',
        color: Colors.blue2,
        fontWeight: 'bold'
    },
});

export default styles
