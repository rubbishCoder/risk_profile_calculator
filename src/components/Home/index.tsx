import { StackActions, useNavigation } from '@react-navigation/native';
import React from 'react';
import { SafeAreaView, Text, TouchableOpacity, View } from 'react-native';
import styles from './styles';
import RV from '../../common/utils/RV';

const Home = () => {
  const navigation = useNavigation<any>();

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.textContainer}>
        <Text testID='INTRO_TEXT' style={styles.text}>
          Click the below button in order to take the assessment to find your 'Risk Profile Score'.
        </Text>
        <TouchableOpacity testID={'START_BUTTON'} style={styles.buttonContainer} activeOpacity={1}
          onPress={() => { navigation.dispatch(StackActions.replace('Assessment')); }}>
          <Text style={styles.buttonText}>Start</Text>
        </TouchableOpacity>
      </View >
    </SafeAreaView >
  )
}

export default Home;
