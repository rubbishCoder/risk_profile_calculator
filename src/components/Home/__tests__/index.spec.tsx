import { fireEvent, render } from '@testing-library/react-native';
import React from 'react';
import Home from '..';

let mockedNavigationDispatch = jest.fn();

jest.mock('@react-navigation/native', () => {
    return {
        useNavigation: () => ({
            dispatch: mockedNavigationDispatch
        }),
        StackActions: {
            replace: jest.fn().mockImplementation(x => ({ ...x, "type": "Navigation/REPLACE" }))
        }
    };
});

describe('Home', () => {
    beforeEach(() => {
        mockedNavigationDispatch.mockClear();
    });

    it('should create new snapshot and check whether it match with the previous snapshot', () => {
        const HomeScreen = render(<Home />);
        const HomeScreenJson = HomeScreen.toJSON();
        expect(HomeScreenJson).toMatchSnapshot();
    });

    it('should render introduction text', () => {
        const HomeScreen = render(<Home />);
        const introText = HomeScreen.getByTestId('INTRO_TEXT')
        expect(introText).toBeTruthy()
    });


    it('should render Start button', () => {
        const HomeScreen = render(<Home />);
        const startButton = HomeScreen.getByTestId('START_BUTTON')
        expect(startButton).toBeTruthy()
    });

    it('should navigate when Start button is pressed', () => {
        const HomeScreen = render(<Home />);
        const startButton = HomeScreen.getByTestId('START_BUTTON')
        fireEvent.press(startButton);
        expect(mockedNavigationDispatch).toHaveBeenCalled();
    });
})