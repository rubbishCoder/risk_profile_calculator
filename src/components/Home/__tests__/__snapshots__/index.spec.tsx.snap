// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should create new snapshot and check whether it match with the previous snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fe0133",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 35,
        "fontWeight": "bold",
        "padding": 23,
        "textAlign": "center",
      }
    }
    testID="INTRO_TEXT"
  >
    Click the below button in order to take the assessment to find your 'Risk Profile Score'.
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#3b393a",
        "borderRadius": 17,
        "justifyContent": "center",
        "opacity": 1,
        "paddingHorizontal": 29,
        "paddingVertical": 23,
      }
    }
    testID="START_BUTTON"
  >
    <Text
      style={
        {
          "color": "#ffffff",
          "fontSize": 35,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      Start
    </Text>
  </View>
</RCTSafeAreaView>
`;
